azure-iot-device/tests/common/pipeline/test_pipeline_ops_base.py

```diff
@@ -3,124 +3,47 @@
-
-
-@pytest.mark.describe("Reconnect object")
-class TestReconnect(object):
-    @pytest.mark.it("Sets name attribute on instantiation")
-    @pytest.mark.it("Sets error attribute to None on instantiation")
-    @pytest.mark.it("Sets needs_connection attribute to False on instantiation")
-    @pytest.mark.it("Sets callback attribute to None if not provided on instantiation")
-    def test_required_arguments(self):
-        obj = pipeline_ops_base.Reconnect()
-        assert_all_base_defaults(obj, False)
-        assert obj.callback is None
-
-    @pytest.mark.it("Sets callback attribute if provided on instantiation")
-    def test_optional_arguments(self):
-        obj = pipeline_ops_base.Reconnect(callback=fake_callback)
-        assert obj.callback is fake_callback
-
-
-@pytest.mark.describe("EnableFeature object")
-class TestEnableFeature(object):
-    @pytest.mark.it("Sets name attribute on instantiation")
-    @pytest.mark.it("Sets error attribute to None on instantiation")
-    @pytest.mark.it("Sets needs_connection attribute to False on instantiation")
-    @pytest.mark.it("Sets feature_name attribute on instantiation")
-    @pytest.mark.it("Sets callback attribute to None if not provided on instantiation")
-    def test_required_arguments(self):
-        obj = pipeline_ops_base.EnableFeature(feature_name=fake_feature_name)
-        assert_all_base_defaults(obj, True)
-        assert obj.callback is None
-        assert obj.feature_name is fake_feature_name
-
-    @pytest.mark.it("Sets callback attribute if provided on instantiation")
-    def test_optional_arguments(self):
-        obj = pipeline_ops_base.EnableFeature(
-            feature_name=fake_feature_name, callback=fake_callback
-        )
-        assert obj.callback is fake_callback
-
-
-@pytest.mark.describe("DisableFeature object")
-class TestDisableFeature(object):
-    @pytest.mark.it("Sets name attribute on instantiation")
-    @pytest.mark.it("Sets error attribute to None on instantiation")
-    @pytest.mark.it("Sets needs_connection attribute to False on instantiation")
-    @pytest.mark.it("Sets feature_name attribute on instantiation")
-    @pytest.mark.it("Sets callback attribute to None if not provided on instantiation")
-    def test_required_arguments(self):
-        obj = pipeline_ops_base.DisableFeature(feature_name=fake_feature_name)
-        assert_all_base_defaults(obj, True)
-        assert obj.callback is None
-        assert obj.feature_name is fake_feature_name
-
-    @pytest.mark.it("Sets callback attribute if provided on instantiation")
-    def test_optional_arguments(self):
-        obj = pipeline_ops_base.DisableFeature(
-            feature_name=fake_feature_name, callback=fake_callback
-        )
-        assert obj.callback is fake_callback
+from tests.common.pipeline import pipeline_data_object_test
+
+this_module = sys.modules[__name__]
+
+pipeline_data_object_test.add_operation_test(
+    obj=pipeline_ops_base.PipelineOperation,
```

Comment:
>PipelineOperation [](start = 26, length = 17)

This one should be abstract right so directly not under test?

Reply:
Yes.  Will fix.

---
In reply to: [291707448](https://github.com/Azure/azure-iot-sdk-python-preview/pull/104#discussion_r291707448) [](ancestors = 291707448)